sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -191,9 +191,11 @@ case class DataSource(
     val dataSchema = userSpecifiedSchema.map { schema =>
       StructType(schema.filterNot(f => partitionSchema.exists(p => equality(p.name, f.name))))
     }.orElse {
+      // Remove "path" option so that it is not added to the paths returned by
+      // `tempFileIndex.allFiles()`.
       format.inferSchema(
         sparkSession,
-        caseInsensitiveOptions,
+        caseInsensitiveOptions - "path",
```

Comment:
Fixing at the caller seems good. But `inferSchema` is actually implemented by various format, we don't know if they will use `path` in the implementation. That's one concern.

Reply:
`FileFormat` is an internal API, I think it's natural to assume `paths` parameter will be used, and all builtin implementations do.